### PR TITLE
modify some code

### PR DESCRIPTION
**Pull request by masatodejima0413**

### CODE REVIEW

File: components/App.tsx [hunk 1]
```diff
@@ -8,50 +9,43 @@ export interface IItem {
...
-  const submit = () => {
-    const newItem = createItem();
-    Items.add(newItem).then(docRef => {
-      setItems(prev => [...prev, { ...newItem, key: docRef.id }]);
-    });
-    setVal('');
+  const addItem = () => {
+    const id = uuid();
+    const newItem = createItem(id);
+    setItems(prev => [...prev, newItem as IItem]);
```
Comment (makotodejima) [inline components/App.tsx:31]: why need `as IItem` ? 
this should be naturally `IItem` shouldn't it?

File: pages/index.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ const Home = () => {
           .get()
           .then(snapshot =>
             snapshot.forEach(doc => {
-              setItems(prev => [...prev, { ...doc.data(), key: doc.id } as IItem]);
+              setItems(prev => [...prev, doc.data() as IItem]);
```
Comment (makotodejima) [inline pages/index.tsx:21]: here we probably need `as IItem` because ts doesn't know `doc.data()` is item

Comment (makotodejima) [reply pages/index.tsx:21]: Basically when we use `as [sometype]` (this is called casting). we should really consider if it's really needed. 
Best case we don't want to do any casting. In this case ideally we should do deserialisation or like

**Comment by masatodejima0413** [comment]
@makotodejima 

`export interface IItem {
  createdAt: firebase.firestore.Timestamp;
  text: string;
  userId: string;
  id: string;
  children: string[];
}`
チルドレンたちも一応つくった、add, delete, updateできる

今のところはローカルだけで動く、dbへの追加とかはしてない

**Pull request merged**